flink-ml-lib/src/test/java/org/apache/flink/ml/recommendation/SwingTest.java

```diff
@@ -0,0 +1,249 @@
+import org.apache.flink.types.Row;
+
+import org.apache.commons.collections.IteratorUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests {@link Swing}.
+ */
+public class SwingTest {
+    @Rule
+    public final TemporaryFolder tempFolder = new TemporaryFolder();
+    private StreamExecutionEnvironment env;
+    private StreamTableEnvironment tEnv;
+    private Table trainData;
+
+    @Before
+    public void before() {
+        env = TestUtils.getExecutionEnvironment();
+        tEnv = StreamTableEnvironment.create(env);
+        List<Row> trainRows =
+                new ArrayList<>(
+                        Arrays.asList(
+                                Row.of(0L, 10L),
+                                Row.of(0L, 11L),
+                                Row.of(0L, 12L),
+                                Row.of(1L, 13L),
+                                Row.of(1L, 12L),
+                                Row.of(2L, 10L),
+                                Row.of(2L, 11L),
+                                Row.of(2L, 12L),
+                                Row.of(3L, 13L),
+                                Row.of(3L, 12L),
+                                Row.of(4L, 12L),
+                                Row.of(4L, 10L),
+                                Row.of(4L, 11L),
+                                Row.of(4L, 12L),
+                                Row.of(4L, 13L)));
+        trainData =
+                tEnv.fromDataStream(
+                        env.fromCollection(
+                                trainRows,
+                                new RowTypeInfo(
+                                        new TypeInformation[] {
+                                                BasicTypeInfo.LONG_TYPE_INFO,
+                                                BasicTypeInfo.LONG_TYPE_INFO
+                                        },
+                                        new String[] {"user_id", "item_id"})));
+    }
+
+    private void compareResultAndExpected(List<Row> results) {
+        List<Row> expectedScoreRows =
+                new ArrayList<>(
+                        Arrays.asList(
+                                Row.of(10L, "11,0.058845768947156235;12,0.058845768947156235"),
+                                Row.of(11L, "10,0.058845768947156235;12,0.058845768947156235"),
+                                Row.of(
+                                        12L,
+                                        "13,0.09134833828228624;10,0.058845768947156235;11,0.058845768947156235"),
+                                Row.of(13L, "12,0.09134833828228624")));
+
+        results.sort((o1, o2) -> Long.compare(o1.getFieldAs(0), o2.getFieldAs(0)));
```

Comment:
nit: This could be simplified as `results.sort(Comparator.comparingLong(o -> o.getFieldAs(0)));`